augur/tasks/git/facade_tasks.py

```diff
@@ -282,13 +279,54 @@ def analyze_commits_in_parallel(repo_git, multithreaded: bool)-> None:
 
 
             #logger.info(f"Got to analysis!")
-            absoulte_path = get_absolute_repo_path(session.repo_base_directory, repo.repo_id, repo.repo_path,repo.repo_name)
-            repo_loc = (f"{absoulte_path}/.git")
+            commitRecord = analyze_commit(session, repo_id, repo_loc, commitTuple)
+            #logger.debug(commitRecord)
+            if commitRecord:
+                if len(pendingCommitRecordsToInsert) < 10000:
+                    pendingCommitRecordsToInsert.append(commitRecord)
+                else:
+
+                    session.execute(
+                        insert(Commit),
+                        pendingCommitRecordsToInsert,
+                    )
+                    session.commit()
+                    #session.insert_data(pendingCommitRecordsToInsert, Commit,["repo_id","cmt_commit_hash","cmt_filename", "cmt_committer_date"],on_conflict_update=False)
+                    pendingCommitRecordsToInsert = []
+
+        try:
+            session.execute(
+                    insert(Commit),
+                    pendingCommitRecordsToInsert,
+                )
+            session.commit()
+        except DataError as e:
+            session.logger.error(f"Ran into bad data when trying to insert commit with values: \n {commit_record} \n Error: {e}")
+
+            #Check for improper utc timezone offset
+            #UTC timezone offset should be betwen -14:00 and +14:00
+
+            for commit_record in pendingCommitRecordsToInsert:
+                if "time zone displacement" in f"{e}":
+                    commit_record['author_timestamp'] = placeholder_date
+                    commit_record['committer_timestamp'] = placeholder_date
```

Comment:
Is it possible to detect an invalid timezone offset before doing the database insert? If so we should detect it in the analyze_commit method and have it update the timestamps to be the placeholder_date.

Reply:
It should be possible from what I remember. You're right this change should be made.